state/disabled/disabledDataTrieHandler.go

```diff
@@ -0,0 +1,39 @@
+package disabled
+
+import (
+	"context"
+
+	"github.com/ElrondNetwork/elrond-go/common"
+)
+
+type disabledDataTrieHandler struct {
+}
+
+// NewDisabledDataTrieHandler returns a new instance of disabledDataTrieHandler
+func NewDisabledDataTrieHandler() *disabledDataTrieHandler {
+	return &disabledDataTrieHandler{}
+}
+
+// RootHash returns an empty byte array
+func (ddth *disabledDataTrieHandler) RootHash() ([]byte, error) {
+	return []byte{}, nil
+}
+
+// GetAllLeavesOnChannel does nothing for this implementation
+func (ddth *disabledDataTrieHandler) GetAllLeavesOnChannel(
+	leavesChannels *common.TrieIteratorChannels,
+	_ context.Context,
+	_ []byte,
+	_ common.KeyBuilder,
+) error {
+	if leavesChannels.LeavesChan != nil {
+		close(leavesChannels.LeavesChan)
```

Comment:
very good catch :+1: the structure has to handle the channel's closing. 
might close the errors chan as well?

Reply:
Done.